core/src/main/java/com/microsoft/applicationinsights/ByteBufferPool.java

```diff
@@ -0,0 +1,29 @@
+package com.microsoft.applicationinsights;
+
+import java.nio.ByteBuffer;
+import java.util.List;
+import java.util.Queue;
+import java.util.concurrent.ArrayBlockingQueue;
+
+class ByteBufferPool {
+
+    private static final int BYTE_BUFFER_SIZE = 65536;
+    private static final int MAX_RETAINED = 10;
```

Comment:
is this too small? the old channel buffer supports up to 1000.

Reply:
this is still 650kb of memory, I think that should be enough for most requests?

Reply:
for perf counters metric type, it collects every mins.. i have seen average up to 80+ telemetries per request.

![image](https://user-images.githubusercontent.com/56097766/119575959-ac506680-bd6c-11eb-806d-f583321760b6.png)

Reply:
multiple telemetries will fit in one bytebuffer